integration/spa-mode-test.ts

```diff
@@ -308,152 +310,232 @@ test.describe("SPA Mode", () => {
-      let app = new PlaywrightFixture(appFixture, page);
-      await app.goto("/");
-      expect(await page.locator("[data-route]").textContent()).toBe("Index");
+      test("errors if you call serverLoader", async ({ page }) => {
+        let app = new PlaywrightFixture(appFixture, page);
+        await app.goto("/");
+        expect(await page.locator("[data-route]").textContent()).toBe("Index");
 
-      await app.clickLink("/about");
-      await page.waitForSelector('[data-route]:has-text("About")');
-      expect(await page.locator("[data-route]").textContent()).toBe("About");
-      expect(await page.locator("[data-loader-data]").textContent()).toBe(
-        "About Loader Data"
-      );
-      expect(await page.locator("title").textContent()).toBe(
-        "About Title: About Loader Data"
-      );
-    });
+        await app.clickLink("/error");
+        await page.waitForSelector("[data-error]");
+        expect(await page.locator("[data-error]").textContent()).toBe(
+          'Error: You cannot call serverLoader() in SPA Mode (routeId: "routes/error")'
+        );
+      });
 
-    test("navigates and calls actions/loaders", async ({ page }) => {
-      let app = new PlaywrightFixture(appFixture, page);
-      await app.goto("/");
-      expect(await page.locator("[data-route]").textContent()).toBe("Index");
+      test("errors if you call serverAction", async ({ page }) => {
+        let app = new PlaywrightFixture(appFixture, page);
+        await app.goto("/");
+        expect(await page.locator("[data-route]").textContent()).toBe("Index");
 
-      await app.clickSubmitButton("/about");
-      await page.waitForSelector('[data-route]:has-text("About")');
-      expect(await page.locator("[data-route]").textContent()).toBe("About");
-      expect(await page.locator("[data-action-data]").textContent()).toBe(
-        "About Action Data"
-      );
-      expect(await page.locator("[data-loader-data]").textContent()).toBe(
-        "About Loader Data"
-      );
-      expect(await page.locator("title").textContent()).toBe(
-        "About Title: About Loader Data"
-      );
+        await app.clickSubmitButton("/error");
+        await page.waitForSelector("[data-error]");
+        expect(await page.locator("[data-error]").textContent()).toBe(
+          'Error: You cannot call serverAction() in SPA Mode (routeId: "routes/error")'
+        );
+      });
     });
+  });
 
-    test("errors if you call serverLoader", async ({ page }) => {
-      let app = new PlaywrightFixture(appFixture, page);
-      await app.goto("/");
-      expect(await page.locator("[data-route]").textContent()).toBe("Index");
+  test.describe("One-off builds", () => {
+    test("writes out a JS file in library mode", async ({ page }) => {
+      fixture = await createFixture({
+        compiler: "vite",
+        spaMode: true,
+        library: "app.js",
+        files: {
+          "vite.config.ts": js`
```

Comment:
This test shows the basic public-facing usage for library mode:

* set `unstable_library: "app.js"` in your vite plugin config
* `root.tsx` should return a subset of document HTML, not an `<html>` document
* change `entry.client.tsx` to `createRoot(div).render()`, not `hydrateRoot(document)`
* then, serve your library within some other HTML doc (this is handled internally by our test fixture)
  * `<div id="app"/><script type="module" src="/assets/app.js"></script>`